### PR TITLE
Use latest Flutter stable and refactor SDK version parameters

### LINKED ISSUE #8: CircleCI flutter orb unexpected argument: version
## Orb version:
1.0.1

## What happened:
I've got the error:
```
Error calling workflow: 'distribute'Error calling job: 'distribute'Error calling command: 'flutter/install_sdk_and_pub'Unexpected argument(s): version
```

## Expected behavior:
It should download and install Flutter version passed as argument version.

## Additional Information:
Yaml file:

```
version: '2.1'
orbs:
  flutter: circleci/flutter@1.0.1
jobs:
  distribute:
    macos:
      xcode: 12.5.1
    steps:
      - checkout
      - flutter/install_sdk_and_pub:
          version: 2.8.1
          app-dir: ./app
      - flutter/install_ios_pod:
          app-dir: ./app
      - flutter/install_ios_gem:
          app-dir: ./app
      - run:
          command: bundle exec fastlane ios distribute
          working_directory: ios
workflows:
  distribute:
    jobs:
      - distribute
```

### LINKED ISSUE #13: CircleCI outdated usage examples
If someone tries to copy the exact usage examples, builds will fail. For example, not adding the `checkout` step will result in a vague error related to the caching mechanism not being able to check the `pubspec.yaml` of the project file.

### LINKED ISSUE #14: Set default Flutter SDK to Flutter 3 stable
## Describe Request:
Set the default SDK to Flutter 3.0 stable

## Supporting Documentation Links:
https://docs.flutter.dev/development/tools/sdk/releases?tab=macos

### PR DESCRIPTION
**Pull request by joshuadeguzman**
**SEMVER Update Type:**
- [ ] Major
- [ ] Minor
- [ ] Patch

## Description:

The [docs](https://circleci.com/developer/orbs/orb/circleci/flutter#usage-install_for_macos) for this orb is a bit outdated and might confused newcomers trying out this orb, for example, the version parameter issues.

I fixed the `version` requirement and make use of `flutter_version` wherever the Flutter SDK needs to be set.

Next, changed the usage examples with the updated parameters and missing checkout (causes cache key issues).

Last, I updated the default version to the latest stable [Flutter SDK release](https://docs.flutter.dev/development/tools/sdk/releases?tab=macos), including the sample app working on the latest SDK.

## Motivation:

- Closes #14
- Closes #13
- Closes #8

## Checklist:

- [X] All new jobs, commands, executors, parameters have descriptions.
- [X] Usage Example version numbers have been updated.
- [X] Changelog has been updated.

### CODE REVIEW

**Comment by joshuadeguzman** [comment]
Hey guys, would you be able to take a look at this? @tadashi0713

**Comment by JalexChen** [comment]
hey @joshuadeguzman - thank you for putting this PR in. at first glance, this looks alright, but we are holding off on releasing any orb updates at the moment. will revisit this in a bit and let the job run

**Comment by JalexChen** [comment]
hi @joshuadeguzman - do you mind rebasing this on main so we can consolidate and squash commits?

File: src/jobs/lint.yml [hunk 1]
```diff
@@ -1,10 +1,10 @@
 description: >
    Run static analysis(flutter analyze)
 parameters:
-  version:
+  flutter-version:
```
Comment (JalexChen) [inline src/jobs/lint.yml:4]: you will need to change line 17 to be `flutter-version` instead of `version`

File: src/jobs/unit_test.yml [hunk 2]
```diff
@@ -1,10 +1,10 @@
 description: >
   Run Unit Test
 parameters:
-  version:
+  flutter-version:
```
Comment (JalexChen) [inline src/jobs/unit_test.yml:4]: you will need to change line 17 to be `flutter-version`

File: src/commands/install_sdk.yml [hunk 3]
```diff
@@ -1,10 +1,10 @@
 description: >
   Install Flutter SDK
 parameters:
-  version:
+  flutter-version:
```
Comment (JalexChen) [inline src/commands/install_sdk.yml:4]: replace all `<<parameters.version>>` with `<<parameters.flutter-version>>`

**Review by JalexChen — CHANGES_REQUESTED** [review]
again, thank you for the contribution. i would update the PR again with the latest versions if you so choose, but i only ask that you rebase it. 

if you don't update the versions, don't worry we can do that in a separate PR, however, the change in parameter naming is necessary

**Comment by JalexChen** [comment]
Closing this out as the changes made would need to be added with @joshuadeguzman accepting any PRs. Instead, a lot of the changes here are now in PR #23 so that he still gets contributor credit

**Pull request closed without merging**